modules/cli/src/main/scala/scala/cli/commands/DependencyUpdate.scala

```diff
@@ -0,0 +1,122 @@
+
+    val scopedSources = crossSources.scopedSources(buildOptions).orExit(logger)
+
+    def generateActionableUpdateDiagnostic(scope: Scope)
+      : Seq[ActionableDependencyUpdateDiagnostic] = {
+      val sources = scopedSources.sources(scope, crossSources.sharedOptions(buildOptions))
+
+      if (verbosity >= 3)
+        pprint.err.log(sources)
+
+      val options = buildOptions.orElse(sources.buildOptions)
+      ActionableDependencyHandler.createActionableDiagnostics(options).orExit(logger)
+    }
+
+    val actionableMainUpdateDiagnostics = generateActionableUpdateDiagnostic(Scope.Main)
+    val actionableTestUpdateDiagnostics = generateActionableUpdateDiagnostic(Scope.Test)
+    val actionableUpdateDiagnostics =
+      (actionableMainUpdateDiagnostics ++ actionableTestUpdateDiagnostics).distinct
+
+    if (options.all)
+      updateDependencies(actionableUpdateDiagnostics, logger)
+    else {
+      println("Updates")
+      actionableUpdateDiagnostics.foreach(update =>
+        println(s"   * ${update.oldDependency.render} -> ${update.newVersion}")
+      )
+      println("""|To update all dependencies run: 
+                 |    scala-cli dependency-update --all""".stripMargin)
+    }
+  }
+
+  private def updateDependencies(
+    actionableUpdateDiagnostics: Seq[ActionableDependencyUpdateDiagnostic],
+    logger: Logger
+  ): Unit = {
+    val groupedByFileDiagnostics =
+      actionableUpdateDiagnostics.flatMap {
+        diagnostic =>
+          diagnostic.positions.collect {
+            case file: Position.File =>
+              file.path -> (file, diagnostic)
+          }
+      }.groupMap(_._1)(_._2)
+
+    groupedByFileDiagnostics.foreach {
+      case (Right(file), diagnostics) =>
+        val sortedByLine       = diagnostics.sortBy(_._1.startPos._1).reverse
+        val appliedDiagnostics = updateDependencies(file, sortedByLine)
+        os.write.over(file, appliedDiagnostics)
+        diagnostics.foreach(diagnostic =>
+          logger.message(s"Updated dependency to: ${diagnostic._2.to}")
+        )
+      case (Left(file), diagnostics) =>
+        diagnostics.foreach {
+          diagnostic =>
+            logger.message(s"Warning: Scala CLI can't update ${diagnostic._2.to} in $file")
+        }
+    }
+  }
+
+  private def updateDependencies(
+    file: Path,
+    diagnostics: Seq[(Position.File, ActionableDependencyUpdateDiagnostic)]
+  ): String = {
+    val fileContent   = os.read(file)
+    val startIndicies = Position.Raw.lineStartIndices(fileContent)
+
+    diagnostics.foldLeft(fileContent) {
+      case (fileContent, (file, diagnostic)) =>
+        val (line, column) = (file.startPos._1, file.startPos._2)
+        val startIndex = {
+          val index = startIndicies(line) + column
+          val skipIvySyntax =
+            fileContent.slice(index, index + 6) match {
+              case "$ivy.`" | "$dep.`" => 6
+              case _                   => 0
+            }
+          index + skipIvySyntax
```

Comment:
The skip ivy stuff should be handled from `ActionableDependencyHandler` IMO. Downstream users of it shouldn't have to mind about it.

Reply:
In `ActionableDependencyHandler` I don't have information about type of dependency.  

For example, `Positioned[AnyDependency]` contains only this informations:
```
Positioned(List(File(Right(/scala-cli/Hello.scala),(0,13),(0,37))),com.lihaoyi::geny:0.6.5)
```

so I change resolving ivy dependencies and update start position in last commit.